test/exe/signals_test.cc

```diff
@@ -0,0 +1,96 @@
+#include <signal.h>
+#include <sys/mman.h>
+
+#include "exe/signal_action.h"
+
+#include "gtest/gtest.h"
+
+#if defined(__has_feature)
+#if __has_feature(address_sanitizer)
+#define ASANITIZED /* Sanitized by Clang */
+#endif
+#endif
+
+#if defined(__SANITIZE_ADDRESS__)
+#define ASANITIZED /* Sanitized by GCC */
+#endif
+
+// Memory violation signal tests are disabled under address sanitizer.  The
+// sanitizer does its own special signal handling and prints messages that are
+// not ours instead of what this test expects.  The signals special handled by ASAN
+// include SIGSEGV, SIGBUS, and SIGFPE.
+#ifndef ASANITIZED
+TEST(Signals, InvalidAddressDeathTest) {
+  SignalAction actions;
+  EXPECT_DEATH([]() -> void {
```

Comment:
That's an interesting idea.  It still isn't compatible with the ASAN issue, which seems to inhibit normal signal handling entirely, but it would make the test more robust.  I might implement the test as a "fun project" in a separate PR when I have some hacking time but if you're OK with the current state of the test I'm good with merging it as it stands.